### PR TITLE
Changes for making multi-arch targets

### PR DESCRIPTION
**Pull request by Pensu**
This PR ensure multi-arch images builds for rescheduler.

### CODE REVIEW

**Comment by piosz** [comment]
LGTM from me but I'll let @luxas to take a look as well.

File: rescheduler/Makefile [hunk 1]
```diff
@@ -1,25 +1,65 @@
-all: build
 
 FLAGS =
 ENVVAR = GOOS=linux GOARCH=amd64 CGO_ENABLED=0
 REGISTRY = gcr.io/google-containers
 TAG = v0.3.1
+ARCH ?= $(shell go env GOARCH)
+ALL_ARCH = amd64 arm arm64 ppc64le
```
Comment (luxas) [inline rescheduler/Makefile]: s390x?

File: rescheduler/Makefile [hunk 2]
```diff
@@ -1,25 +1,65 @@
...
+
+
+IMAGE = $(REGISTRY)/rescheduler
+MULTI_ARCH_IMG = $(IMAGE)-$(ARCH)
+
+BASEIMAGE ?= busybox:latest
+
+ifeq ($(ARCH),arm)
+        BASEIMAGE=armhf/busybox:latest
```
Comment (luxas) [inline rescheduler/Makefile]: now arm32v7/

File: rescheduler/Makefile [hunk 3]
```diff
@@ -1,25 +1,65 @@
...
+        BASEIMAGE=ppc64le/busybox:latest
+endif
 
 deps:
 	go get github.com/tools/godep
 
 build: clean deps
-	$(ENVVAR) godep go build ./...
-	$(ENVVAR) godep go build -o rescheduler
+	GOOS=linux GOARCH=$(ARCH) CGO_ENABLED=0 godep go build ./...
```
Comment (luxas) [inline rescheduler/Makefile]: is godep really needed anymore?

**Review by luxas — COMMENTED** [review]
Nearly LGTM

File: rescheduler/Makefile (hunk 3 above)
Comment (mkumatag) [reply rescheduler/Makefile]: I do agree, if godep is not required then we can remove godep dependencies.

File: rescheduler/Makefile (hunk 1 above)
Comment (Pensu) [reply rescheduler/Makefile]: Added it.

File: rescheduler/Makefile (hunk 2 above)
Comment (Pensu) [reply rescheduler/Makefile]: Changed!

File: rescheduler/Makefile (hunk 3 above)
Comment (Pensu) [reply rescheduler/Makefile]: Godep was there earlier. Reran the build after removing it, was able to run it successfully, guess we don't need it, have removed it!

**Review by luxas — APPROVED** [review]

**Comment by luxas** [comment]
@piosz Will you push the images? Thanks!

**Comment by piosz** [comment]
I'm doing it right now.

**Comment by piosz** [comment]
I pushed the following images:
```
gcr.io/google-containers/rescheduler-s390x:v0.3.2
gcr.io/google-containers/rescheduler-ppc64le:v0.3.2
gcr.io/google-containers/rescheduler-arm64:v0.3.2
gcr.io/google-containers/rescheduler-arm:v0.3.2
gcr.io/google-containers/rescheduler-amd64:v0.3.2
```

**Pull request merged**